modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/planner/JoinColocationPlannerTest.java

```diff
@@ -70,10 +78,105 @@ public void joinSameTableSimpleAff() throws Exception {
     }
 
     /**
-     * Join of the same tables with a complex affinity is expected to be colocated.
+     * Join of the same tables with a simple affinity is expected to be colocated.
+     */
+    @Test
+    public void joinSameTableSimpleAffHashJoin() throws Exception {
+        IgniteTable tbl = simpleTable("TEST_TBL", DEFAULT_TBL_SIZE);
+
+        IgniteSchema schema = createSchema(tbl);
+
+        String sql = "select count(*) "
+                + "from TEST_TBL t1 "
+                + "join TEST_TBL t2 on t1.id = t2.id";
+
+        // Only hash join
+        RelNode phys = physicalPlan(sql, schema, "NestedLoopJoinConverter", "CorrelatedNestedLoopJoin", "MergeJoinConverter");
+
+        AbstractIgniteJoin join = findFirstNode(phys, byClass(AbstractIgniteJoin.class));
+        List<RelNode> joinNodes = findNodes(phys, byClass(AbstractIgniteJoin.class));
+
+        String invalidPlanMsg = "Invalid plan:\n" + RelOptUtil.toString(phys);
+
+        assertThat(invalidPlanMsg, joinNodes.size(), equalTo(1));
+        assertThat(invalidPlanMsg, join, notNullValue());
+        assertThat(invalidPlanMsg, join.distribution().function().affinity(), is(true));
+    }
+
+    /**
+     * Hash join need to preserve left collation.
      */
     @Test
-    public void joinSameTableComplexAff() throws Exception {
+    public void hashJoinCheckLeftCollationsPropagation() throws Exception {
```

Comment:
better to move this test to HashJoinPlannerTest

Reply:
done